core/store/src/flat/storage.rs

```diff
@@ -216,44 +234,87 @@ impl FlatStorage {
     /// Update the head of the flat storage, including updating the flat state in memory and on disk
     /// and updating the flat state to reflect the state at the new head. If updating to given head is not possible,
     /// returns an error.
-    pub fn update_flat_head(&self, new_head: &CryptoHash) -> Result<(), FlatStorageError> {
+    /// If `strict`, then unconditionally sets flat head to the given block.
+    /// If not `strict`, then uses `new_head` as a hint and may set flat head to
+    /// `new_head` or to a predecessor of `new_head`.
```

Comment:
I'd like to have this part documented more strictly, to simplify onboarding of new devs to flat storage. Ideally it shouldn't contain a word "final" because flat storage doesn't know anything about finality which is a chain concept.

Now, though I understand that blocks with state changes are important, I can't formulate what are the guarantees of non-strict mode.

Reply:
Rewrote to define the guarantees more strictly.
```
    /// If not `strict`, then it updates the flat head to the **oldest** block X
    /// such that [X, block_hash] contains at most 2 blocks with flat state changes.
```